File: server/services/openaiService.js
// server/services/openaiService.js
const { OpenAI } = require("openai");
require("dotenv").config();

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

exports.getChatCompletion = async (resume, jobDescription) => {
  const messages = [
    {
      role: "system",
      content: "You are a resume reviewer that compares resumes with job descriptions and gives helpful feedback.",
    },
    {
      role: "user",
      content: `Resume:\n${resume}\n\nJob Description:\n${jobDescription}`,
    },
  ];

  try {
    const completion = await openai.chat.completions.create({
      model: "gpt-3.5-turbo", // or gpt-4 if you have access
      messages,
      temperature: 0.7,
      max_tokens: 800,
    });

    return completion.choices[0].message.content;
  } catch (err) {
    console.error("OpenAI Error:", err.message);
    throw new Error("Failed to get response from OpenAI.");
  }
};

File: server2/services/openai_service.py
import os
from openai import AzureOpenAI
from dotenv import load_dotenv

load_dotenv()

client = AzureOpenAI(
    api_key=os.getenv("AZURE_OPENAI_KEY"),
    api_version=os.getenv("AZURE_OPENAI_VERSION"),
    azure_endpoint=os.getenv("AZURE_OPENAI_ENDPOINT")
)

deployment_name = os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME")

def analyze_resume(resume_text: str, job_description: str) -> str:
    messages = [
        {
            "role": "system",
            "content": "You are a resume reviewer that compares resumes with job descriptions and gives helpful feedback.",
        },
        {
            "role": "user",
            "content": resume_text,
        },
        {
            "role": "user",
            "content": job_description,
        },
        {
            "role": "assistant",
            "content": "Please analyze the resume against the job description and provide feedback on how well the resume matches the job requirements, including strengths and areas for improvement.",
        }
    ]

    response = client.chat.completions.create(
        model=deployment_name,
        messages=messages,
        temperature=0.7,
        max_tokens=800
    )

    return response.choices[0].message.content
